### PR TITLE
Move to Production Profile Page Fixes

### PR DESCRIPTION
**Pull request by amitojsingh**

### CODE REVIEW

**Review by MySmilingTurban — APPROVED** [review]
Looks good to me!

File: src/components/buttons/TextToSpeechBtn.tsx [hunk 1]
```diff
@@ -20,12 +23,15 @@ const TextToSpeechBtn: FC<TextToSpeechBtnProps> = ({
...
+  classname = '',
+  rounded = true,
+  backgroundColor = '',
+  divider = '',
   setLoading,
   size = CONSTANTS.TEXT_TO_SPEECH_BTN_SIZE,
 }) => {
   if (!text) {
-    console.log(text, type, audioURL, id, backgroundColor, setLoading, size);
+    console.log(text, type, audioURL, id, classname, rounded, backgroundColor, divider, setLoading, size);
```
Comment (gurjit03) [inline src/components/buttons/TextToSpeechBtn.tsx:34]: @amitojsingh stale log

File: src/components/buttons/TextToSpeechBtn.tsx [hunk 2]
```diff
@@ -35,7 +41,10 @@ const TextToSpeechBtn: FC<TextToSpeechBtnProps> = ({
 //   type,
 //   audioURL,
 //   id,
-//   backgroundColor,
+//   classname = '',
```
Comment (gurjit03) [inline src/components/buttons/TextToSpeechBtn.tsx:44]: @amitojsingh commented code

File: src/components/buttons/tests/OpenFeedbackBtn.test.tsx [hunk 3]
```diff
@@ -1,20 +1,21 @@
...
  */
-import React from 'react';
+import React, { PropsWithChildren } from 'react';
 import { render, fireEvent, screen } from '@testing-library/react';
 import '@testing-library/jest-dom';
 import FeedbackBtn from '../OpenFeedbackBtn';
 
-// Mocking the imported components
 // Mocking the FeedbackForm component
 jest.mock('components/feedback', () => {
```
Comment (gurjit03) [inline src/components/buttons/tests/OpenFeedbackBtn.test.tsx:10]: @amitojsingh just wondering, why we need to mock this component here ?

File: src/components/buttons/tests/OpenFeedbackBtn.test.tsx [hunk 4]
```diff
@@ -1,20 +1,21 @@
...
   const MockFeedbackForm = () => <div data-testid='mock-feedback-form'>Feedback Form</div>;
   MockFeedbackForm.displayName = 'MockFeedbackForm'; // Setting the displayName
   return MockFeedbackForm;
 });
+
+// Mocking the imported components
 jest.mock('components/modal', () => {
-  const MockModal = ({ children }: { children: React.ReactNode }) => (
+  const MockModal = ({ children }: PropsWithChildren) => (
```
Comment (gurjit03) [inline src/components/buttons/tests/OpenFeedbackBtn.test.tsx:18]: @amitojsingh should we include tests to make sure the feedback is submitted correctly with correct arguments also ?

File: src/components/buttons/tests/option.test.tsx [hunk 5]
```diff
@@ -76,7 +76,7 @@ describe('OptionBtn', () => {
         disabled={false}
       />,
     );
-    expect(screen.getByRole('button').parentNode).toHaveClass('bg-lightGreen');
+    expect(screen.getByRole('button').parentElement).toHaveClass('bg-lightGreen');
```
Comment (gurjit03) [inline src/components/buttons/tests/option.test.tsx:79]: @amitojsingh I highly recommend using other selector than `parentElement/parentNode` for getting element ji.

File: src/utils/storage.tsx [hunk 6]
```diff
@@ -1,24 +1,14 @@
...
-import { User } from 'types';
+import { User } from 'firebase/auth';
 
-export const uploadImage = async (
-  file: File,
-  currentUser: User,
-  setLoading: Dispatch<boolean>,
-  setPhotoURL: Dispatch<string>,
-) => {
+export const uploadImage = async (file: File, currentUser: User, setLoading: Dispatch<boolean>) => {
```
Comment (gurjit03) [inline src/utils/storage.tsx:6]: @amitojsingh lets remove all the side effects out of this function as it's not this function responsibility to handle the loading. Clearly this function is for uploading the Image. Loading indicator should be handled in the caller.

**Review by gurjit03 — CHANGES_REQUESTED** [review]
Great work @amitojsingh 🔥  some minor comments :)

**Comment by amitojsingh** [comment]
@gurjit03 Veer ji, I'll create a separate PR for the enhancements you suggested and we will merge in the next Release. It's not best practice to push code directly to the main branch. Currently, regression testing is underway. Should we ensure all functionalities in this PR are tested before merging to prod? If we merge additional code into this PR, it could invalidate the testing already completed by everyone.

**Comment by gurjit03** [comment]
> @gurjit03 Veer ji, I'll create a separate PR for the enhancements you suggested and we will merge in the next Release. It's not best practice to push code directly to the main branch. Currently, regression testing is underway. Should we ensure all functionalities in this PR are tested before merging to prod? If we merge additional code into this PR, it could invalidate the testing already completed by everyone.

@amitojsingh yes definitely !

**Review by gurjit03 — APPROVED** [review]

**Pull request merged**